src/main/java/org/opensrp/domain/rapidpro/converter/BaseRapidProEventConverter.java

```diff
@@ -0,0 +1,119 @@
+package org.opensrp.domain.rapidpro.converter;
+
+import org.apache.commons.lang3.StringUtils;
+import org.opensrp.common.util.DateUtil;
+import org.opensrp.domain.postgres.Organization;
+import org.opensrp.domain.rapidpro.contact.zeir.RapidProContact;
+import org.opensrp.domain.rapidpro.converter.zeir.ZeirVaccine;
+import org.opensrp.service.OrganizationService;
+import org.opensrp.util.DateParserUtils;
+import org.opensrp.util.constants.RapidProConstants;
+import org.smartregister.domain.Event;
+import org.smartregister.domain.Obs;
+
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+public abstract class BaseRapidProEventConverter implements RapidProContactEventConverter {
+
+	protected OrganizationService organizationService;
+
+	public BaseRapidProEventConverter() {
+	}
+
+	public BaseRapidProEventConverter(OrganizationService organizationService) {
+		this.organizationService = organizationService;
+	}
+
+	protected void addCommonEventProperties(RapidProContact rapidProContact, Event event) {
+		event.setBaseEntityId(rapidProContact.getUuid());
+		event.setFormSubmissionId(UUID.randomUUID().toString());
+		event.setLocationId(rapidProContact.getFields().getFacilityLocationId());
+		event.setProviderId(rapidProContact.getFields().getSupervisorPhone());
+		Organization organization = organizationService
+				.getOrganizationByLocationId(rapidProContact.getFields().getFacilityLocationId());
```

Comment:
Shouldn't a null check be added for `rapidProContact.getFields()` ?

Reply:
Never expecting a null for the fields. But it is okay to have an extra validation step.